### PR TITLE
Fix/same submission

### PR DESCRIPTION
**Pull request by cyfml**
1. fixed same submission problem which leads Comparision-Viewer broken.
2. fixed issue: The folder in the results holding the submission files should be renamed from "submissions" to "files", and report-viewer also supports old version of Jplag. (e.g. submissions in zip file).
3. Enhancement:  
-  where the files are dropped should contain a line of text that states that no files get uploaded anywhere
-  The code comparison view always shows "submission 1" and "submission 2", but should show the names. But not if the names are hidden.
-  In the code comparison view, the left sidebar should be visible by default

### CODE REVIEW

**Comment by cyfml** [comment]
Now, report-viewer supports old version of Jplag. I also find a bug in the previous PR. This bug is: if root name or sub directory is also called "submissions". The Comparison view will not be broken but show not full path, then it influences the highlighting.
Previous one seems a coincidental, now i fixed this problem.

File: core/src/main/java/de/jplag/reporting/jsonfactory/DirectoryManager.java [hunk 1]
```diff
@@ -34,8 +34,7 @@ public static File createDirectory(String path, String name, File file, File sub
         File directory;
         String fileName = file.getPath();
         String submissionRootPath = submissionRoot.getPath();
-        int lastDirectoryIndex = findRootDirIndex(name, submissionRootPath);
-        fileName = fileName.substring(lastDirectoryIndex).replaceFirst(name, "");
+        fileName = fileName.substring(submissionRootPath.length());
```
Comment (tsaglam) [inline core/src/main/java/de/jplag/reporting/jsonfactory/DirectoryManager.java:37]: If you only want the name, you can fo `file.getName()` instead.

**Review by tsaglam — APPROVED** [review]
Just one small thing.

File: core/src/main/java/de/jplag/reporting/jsonfactory/DirectoryManager.java (hunk 1 above)
Comment (cyfml) [reply core/src/main/java/de/jplag/reporting/jsonfactory/DirectoryManager.java:37]: Actually, it is with a separator, for example, \TerrainType.java or /TerrainType.java.

Comment (tsaglam) [reply core/src/main/java/de/jplag/reporting/jsonfactory/DirectoryManager.java:37]: Then I would suggest `String fileName = File.separator() + file.getName();`. This makes it clearer what is actually required there.

File: report-viewer/src/components/FilesContainer.vue [hunk 2]
```diff
@@ -13,7 +13,9 @@
         :lines="!files.get(file)?.lines ? [] : files.get(file)?.lines"
         :matches="!matches.get(file) ? [] : matches.get(file)"
         :panel-id="containerId"
-        :title="file.length>40?'..'+file.substring(file.length-40,file.length):file"
+        :title="convertSubmissionIdToName(file, submissionId).length > 40?
```
Comment (sebinside) [inline report-viewer/src/components/FilesContainer.vue:16]: I would suggest moving this to an additional function that calls `convertSubmissionIdToName` with the length check of 40 and so on, to minimize the complexity of this section

**Comment by sebinside** [comment]
Besides this change, I have nothing to complain. I'm not happy with the complexity of the `extractFileNameWithFullPath` function but have no better solution. In my testing, everything worked fine. Can be merged after the small change mentioned above.

**Comment by cyfml** [comment]
@tsaglam @sebinside I suggest, it's better to change the name of PR to "Fix/same submission".  Same root problem was postponed.

**Review by sebinside — APPROVED** [review]

**Comment by sebinside** [comment]
Thank you @cyfml, I just merged this PR and updated issue #710. I also added a new strange bug that I discovered while testing (also prior to this PR). Maybe you can also find a solution to this.

**Pull request merged**